learntools/ml-insights/nbs/tut2.ipynb

```diff
@@ -0,0 +1,446 @@
+       "                \n",
+       "            </td>\n",
+       "            <td style=\"padding: 0 0.5em 0 0.5em; text-align: left; border: none;\">\n",
+       "                Fouls Committed\n",
+       "            </td>\n",
+       "        </tr>\n",
+       "    \n",
+       "        <tr style=\"background-color: hsl(0, 100.00%, 87.69%); border: none;\">\n",
+       "            <td style=\"padding: 0 1em 0 0.5em; text-align: right; border: none;\">\n",
+       "                -0.0375\n",
+       "                \n",
+       "                    &plusmn; 0.0919\n",
+       "                \n",
+       "            </td>\n",
+       "            <td style=\"padding: 0 0.5em 0 0.5em; text-align: left; border: none;\">\n",
+       "                Attempts\n",
+       "            </td>\n",
+       "        </tr>\n",
+       "    \n",
+       "        <tr style=\"background-color: hsl(0, 100.00%, 84.94%); border: none;\">\n",
+       "            <td style=\"padding: 0 1em 0 0.5em; text-align: right; border: none;\">\n",
+       "                -0.0500\n",
+       "                \n",
+       "                    &plusmn; 0.0500\n",
+       "                \n",
+       "            </td>\n",
+       "            <td style=\"padding: 0 0.5em 0 0.5em; text-align: left; border: none;\">\n",
+       "                Ball Possession %\n",
+       "            </td>\n",
+       "        </tr>\n",
+       "    \n",
+       "    \n",
+       "    </tbody>\n",
+       "</table>\n",
+       "    \n",
+       "\n",
+       "    \n",
+       "\n",
+       "\n",
+       "    \n",
+       "\n",
+       "    \n",
+       "\n",
+       "    \n",
+       "\n",
+       "    \n",
+       "\n",
+       "    \n",
+       "\n",
+       "    \n",
+       "\n",
+       "\n",
+       "\n"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "execution_count": 54,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import eli5\n",
+    "from eli5.sklearn import PermutationImportance\n",
+    "\n",
+    "perm = PermutationImportance(my_model, random_state=1).fit(val_X, val_y)\n",
+    "eli5.show_weights(perm, feature_names = val_X.columns.tolist())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Interpreting Permutation Importances\n",
+    "\n",
+    "The values towards the top are the most important features, and those towards the bottom matter least.\n",
+    "\n",
+    "The first number in each row is a measure of how much the model performance decreased with a random shuffling. Like most things in data science, there is some randomness to the exact performance change measured from a shuffling a column.  For that reason, our permutation importance calculation is measured from multiple shuffles.  The number after the **±** is a measure of how much performance varied from one-reshuffling to the next.\n",
```

Comment:
It's not clear to me what it means to say that e.g. the model performance "decreased by 0.075" when we shuffled Goal Scored. Accuracy delta?

Reply:
Also, whatever metric the difference is with respect to, it would be interesting to know its original value, to get a sense of the relative magnitude of these changes.

Reply:
This brings up an interesting point. 

I've always thought of these only in terms of relative importance. In DataRobot, we normalized these so the top feature had an importance of 1, because I think it's hard for most people to have an intuition about shuffling values... and the most natural comparisons are between features in a given model.

In theory, this could still be sensitive to choice of loss metric, though I'm skeptical that's an important issue in practice.  

Natural options are
1) Say what metric is being used and what it's baseline value is
2) Say I don't think you should get caught up in exact numbers, and focus just on ordering and relative values
3) Ignore the issue, as done in the current draft.

My current inclination is towards option 2. I assume you prefer option 1?

Reply:
That's an interesting point.

Just to play devil's advocate, the absolute values of the deltas could be interesting as an indication of a model's robustness. I could imagine two models for the same problem with the same accuracy, where model 1 has importance weights `{A: .01, B: .005, C: .001}` and model 2 has weights `{A: .1, B: .05, C: .01}`. The relative importances are the same, but it seems like useful information that, on the whole, model 1 is way more resilient to noise in any individual feature.

But also, I think there's something nice about the intuitive graspability of being able to say that shuffling the Goals feature causes my accuracy to go down by 1%, or whatever.

If it really is something as simple as delta accuracy, then yeah, I'd favour option 1. 

If you don't want to go down that route, maybe it would be best to just normalize them to have max 1.